test-frame-common/src/main/java/io/skodjob/testframe/resources/KubeResourceManager.java

```diff
@@ -168,33 +168,59 @@ public final <T extends HasMetadata> void createResourceWithoutWait(T... resourc
      */
     @SafeVarargs
     public final <T extends HasMetadata> void createResourceWithWait(T... resources) {
-        createResource(true, resources);
+        createResource(true, false, resources);
+    }
+
+    /**
+     * Creates or updates resources and waits for readiness.
+     * @param resources The resources to create.
+     * @param <T> The type of the resources.
+     */
+    @SafeVarargs
+    public final <T extends HasMetadata> void createOrUpdateResourceWithWait(T... resources) {
+        createResource(true, true, resources);
+    }
+
+    /**
+     * Creates or updates resources.
+     * @param resources The resources to create.
+     * @param <T> The type of the resources.
+     */
+    @SafeVarargs
+    public final <T extends HasMetadata> void createOrUpdateResourceWithoutWait(T... resources) {
+        createResource(false, true, resources);
     }
 
     /**
      * Creates resources with or without waiting for readiness.
      * @param waitReady Flag indicating whether to wait for readiness.
+     * @param allowUpdate Flag indicating if update resource is allowed
      * @param resources The resources to create.
      * @param <T> The type of the resources.
      */
     @SafeVarargs
-    private <T extends HasMetadata> void createResource(boolean waitReady, T... resources) {
+    private <T extends HasMetadata> void createResource(boolean waitReady, boolean allowUpdate, T... resources) {
         for (T resource : resources) {
             ResourceType<T> type = findResourceType(resource);
             pushToStack(resource);
 
             if (resource.getMetadata().getNamespace() == null) {
                 LOGGER.info(LoggerUtils.RESOURCE_LOGGER_PATTERN,
-                        "Creating", resource.getKind(), resource.getMetadata().getName());
+                        allowUpdate ? "Creating or Updating" : "Creating", resource.getKind(),
```

Comment:
wouldn't be better to have `Creating` as one operation and `Updating` as second one?  I mean move the logging stuff to lines 220, 222, 238, 240?

Reply:
I don't think so

Reply:
I thought about having two methods - one for creation and one for update and then just encapsulate it. Similarly as it is done in the the KubeClient 🤷

Reply:
I don't think so

Reply:
From my POV it is better to have exact operation logged. Also it is in the same way implemented here https://github.com/skodjob/test-frame/blob/main/test-frame-common/src/main/java/io/skodjob/testframe/clients/KubeClient.java#L204 . Guess it could be unified at least.